sql/catalyst/src/main/java/org/apache/spark/sql/connector/read/SupportsPushDownAggregates.java

```diff
@@ -49,9 +51,10 @@
    * Whether the datasource support complete aggregation push-down. Spark could avoid partial-agg
    * and final-agg when the aggregation operation can be pushed down to the datasource completely.
    *
+   * @param groupAttrs group by attributes of aggregation.
    * @return true if the aggregation can be pushed down to datasource completely, false otherwise.
    */
-  default boolean supportCompletePushDown() { return false; }
+  default boolean supportCompletePushDown(String[] groupAttrs) { return false; }
```

Comment:
Do we need to pass in `groupAttrs` here? I guess we can use pushed down group by in `JDBCScanBuilder`?

Reply:
NVM. I just realized that in your PR https://github.com/apache/spark/pull/35009 you need `supportCompletePushDown` in `translateAggregate`

Reply:
You means could preserves `groupAttrs` here?

Reply:
I thought at the time of calling this `supportCompletePushDown`, we already have pushed down `group by` in  `JDBCScanBuilder` so we can just use that one. Then I realized that in your another PR, you need this `supportCompletePushDown` before the `group by` is pushed down. It seems we have to pass in this `groupAttrs`.

Reply:
Why not just pass `Aggregation`?

Reply:
OK